### ISSUE TITLE
Project Structure

### ISSUE THREAD

**Issue by PandelisZ** [HaCSBCU/MesaHub#55]
Currently it looks a little like this

|auth
|bin
|db
|scripts
|test

I'd like to make it something like

|scripts/ (or lib)
| .. |auth/
| .. | db/
| .. | etc/
|tests/
|bin/
For general readability and ease of imports later on via an optional scripts.js in the root of scripts/

Discuss..

**Comment by PandelisZ** [comment]
Oh this is helpful,

http://stackoverflow.com/questions/5178334/folder-structure-for-a-node-js-project

**Comment by alexwileyy** [comment]
@PandelisZ don't see the need for things to be in a ```scripts``` folder, however I do like the way the StackOver flow suggests:
```
| models
  | user.js
  | events.js
  | announcements.js
  | ...
| views
  | dashboard
  | admin
  | ...
| public
  | js
  | css
  | img
  | ..
| controllers
  | index.js
  | admin.js
  | send-text.js
  | ...
//Additional folder structure I think could be useful
| tests
  | tests...
| middlewares
  | minify-image.js
  | authentication (example - passport reality)
  | ...
```

The above is a common project structure that i've been used in several production products. Up for discussion but I think the above cleans things up a lot and sections purpose to folders 😄

**Comment by PandelisZ** [comment]
Yeah, a middleware folder sounds good. Are middleware and lib an interchangeable title? 

I'm asking because I'm not 100% familiar with project structures

**Comment by alexwileyy** [comment]
Yeah I'm fairly sure they are, a lot of structures technically are the same just use different names :)

I've seen the above structure used quite a bit though, we can implement it once these hackathons are over
